src/main/java/org/jabref/logic/search/indexing/DefaultLinkedFilesIndexer.java

```diff
@@ -315,6 +315,11 @@ public void close() {
         HeadlessExecutorService.INSTANCE.execute(this::closeIndex);
     }
 
+    @Override
+    public void closeAndWait() {
+        HeadlessExecutorService.INSTANCE.executeAndWait(this::closeIndex);
```

Comment:
Can we pass this dependency in the constructor as well?

Reply:
Can you check this commit [`81a30ff` (#11749)](https://github.com/JabRef/jabref/pull/11749/commits/81a30ff445045ce6baf99bd0a3ebf3188bc7b431)? added a boolean parameter to the constructor.

Reply:
Yeah, sorry, I did not clarfiy. I meant the Executor Service, that would partially also improve the test siutation

Reply:
TL;DR: Refactor `DefaultLinkedFilesIndexer` NOT to depend on `HeadlessExecutorService.INSTANCE`, but to have an `ExecutorService` passed using the constructor.

--> follow-up PR